src/main/src/main/java/org/geoserver/catalog/ResourcePool.java

```diff
@@ -1587,6 +1588,13 @@ private GridCoverageReader getGridCoverageReader(
             // GeoServer does not need to be updated to the multicoverage stuff
             // (we might want to introduce a hint later for code that really wants to get the
             // multi-coverage reader)
+
+            if (reader.getFormat() instanceof GeoTiffFormat) { //  GEOS-9236
+                if (coverageInfo.getNativeCoverageName() == "geotiff_coverage") {
```

Comment:
Mind, String comparison in Java shouldn't use "==".
You may need to do a "geotiff_coverage".equals(coverageInfo.getNativeCoverageName()).

having the known string as first object also works as a guard in case the 2nd operand is null.